internal/transport/http2_client.go

```diff
@@ -1684,7 +1688,12 @@ func (t *http2Client) reader(errCh chan<- error) {
 		case *http2.PingFrame:
 			t.handlePing(frame)
 		case *http2.GoAwayFrame:
-			t.handleGoAway(frame)
+			{
```

Comment:
Why the code block here?

Reply:
Because `t.handleGoAway` also calls `t.close` which again poses the same problem.

Reply:
Then...I don't think this does what you think?  A bare code block has no effect beyond scoping variables.

Reply:
I didn't understand? I just want that `t.Close` should be called after the `t.readerDone` is closed. So, I just return the error from `t.handleGoAway` and call close when `error!=null`